instrumentation/elasticsearch/elasticsearch-api-client-7.16/javaagent/src/test/java/ElasticsearchClientTest.java

```diff
@@ -0,0 +1,252 @@
+                    span.hasName("index")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasNoParent()
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "elasticsearch"),
+                            equalTo(SemanticAttributes.DB_OPERATION, "index"),
+                            equalTo(SemanticAttributes.NET_PEER_NAME, httpHost.getHostName()),
+                            equalTo(SemanticAttributes.NET_PEER_PORT, httpHost.getPort()),
+                            equalTo(SemanticAttributes.HTTP_METHOD, "PUT"),
+                            equalTo(
+                                SemanticAttributes.HTTP_URL,
+                                httpHost.toURI() + "/test-index/_doc/test-id?timeout=10s"),
+                            equalTo(
+                                AttributeKey.stringKey("db.elasticsearch.path_parts.index"),
+                                "test-index"),
+                            equalTo(
+                                AttributeKey.stringKey("db.elasticsearch.path_parts.id"),
+                                "test-id")),
+                span ->
+                    span.hasName("PUT")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.NET_PEER_NAME, httpHost.getHostName()),
+                            equalTo(SemanticAttributes.NET_PEER_PORT, httpHost.getPort()),
+                            equalTo(SemanticAttributes.HTTP_METHOD, "PUT"),
+                            equalTo(AttributeKey.stringKey("net.protocol.name"), "http"),
+                            equalTo(AttributeKey.stringKey("net.protocol.version"), "1.1"),
+                            equalTo(
+                                SemanticAttributes.HTTP_URL,
+                                httpHost.toURI() + "/test-index/_doc/test-id?timeout=10s"),
+                            equalTo(SemanticAttributes.HTTP_STATUS_CODE, 201L),
+                            equalTo(SemanticAttributes.USER_AGENT_ORIGINAL, userAgent()),
+                            satisfies(
+                                SemanticAttributes.HTTP_RESPONSE_CONTENT_LENGTH,
+                                AbstractLongAssert::isPositive))));
+  }
+
+  @Test
+  public void elasticsearchStatusAsync() throws Exception {
+    CountDownLatch countDownLatch = new CountDownLatch(1);
+    AsyncRequest request = new AsyncRequest();
+
+    runWithSpan(
+        "parent",
+        () ->
+            asyncClient
+                .info()
+                .thenAccept(
+                    infoResponse ->
+                        runWithSpan(
+                            "callback",
+                            () -> {
+                              countDownLatch.countDown();
+                              request.setResponse(infoResponse);
```

Comment:
Ordering these method in reverse would be safer:
```suggestion
                              request.setResponse(infoResponse);
                              countDownLatch.countDown();
```

Reply:
done